### PR TITLE
SF-1417c Account for text in display-text-anchor when drop

### PR DESCRIPTION
**Pull request by marksvc**
- When dragging to a segment with note embeds, the display-text-anchor
    element isn't just a length 1 item to include in the calculation for
    insertion position. It also contains the text that is anchored to.
    Account for this, and adjust Chromium and Firefox tests to behave like
    the browsers in this regard.
- text.component.spec.ts: Set `textAnchor` `length` to 7 to span the
text 'chapter'.

====

This was fixed on the way to addressing the actual SF-1417 issue.

This PR was previously described as "SF-1417b ...", and contained some refactoring. That refactoring was split out and applied onto master, which was then merged to develop, and this PR was rebased on top of the refactoring.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1179)

### CODE REVIEW

**Review by marksvc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1179)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @irahopkinson)

___
*[src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts, line 59 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-MnUzcqcEVFW-5O9CtVc:-Mnbi8xy5c139CCuLrPc:brt6oay) ([raw file](https://github.com/sillsdev/web-xforge/blob/25e8fea51c6cf6490d1e15c0cc310a8cd1d2fa29/src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts#L59)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
>       // (Rather than use type `number|undefined` and check if insertionPositionInDocument==null, use type `number`
>       // and set an invalid value instead of undefined so that the closure in setTimeout() later has a simpler type to
>       // think about.)
> ```
I think this is probably too hacky when all you need to do is a null check in the arrow function or if you have already done the null check then assert it is defined (but be sure that it really is). Err... where is the `setTimeout`? It looks like you would just return on `undefined` anyway.
</blockquote></details>

Oops. Sorry for the confusion. When I split up my work into smaller chunks for review, I missed this. I am setting this back to a `number|undefined` type and will consider what you're saying for the next change coming.
___
*[src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts, line 183 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-MnV1y5vAhINVTqG7lzW:-Mnbhikv8QGTWXotSx2i:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/25e8fea51c6cf6490d1e15c0cc310a8cd1d2fa29/src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts#L183)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> ?
> ```
I don't think you need this `?`.
</blockquote></details>

Done.
___
*[src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts, line 184 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-MnV2MKG61IrCehfiDB1:-MnbhmgqByjynmS9soAy:bss4a2b) ([raw file](https://github.com/sillsdev/web-xforge/blob/25e8fea51c6cf6490d1e15c0cc310a8cd1d2fa29/src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/drag-and-drop.ts#L184)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> : number
> ```
This is redundant since we know its a number from the literal being assigned.
</blockquote></details>

Removed.

**Review by irahopkinson — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-:-MnxkgNI0BV_blvzhE2-:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @marksvc)

**Review by irahopkinson — APPROVED** [review]
I haven't merged this because I wonder if this refactor should really be done on master first and then merge master across to develop? Feel free to merge this as is if it doesn't fit with what you are working on.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-:-MnxktBk3Ih6YIJaOWzh:bnkwutx)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @marksvc)

**Comment by marksvc** [comment]
Good idea. I plan to open a new PR with just the refactor part (since display-text-anchor is on develop but not master).

**Comment by marksvc** [comment]
Ok this PR is waiting on: 
- #1184 merging
- master merging to develop
- update this PR so it includes its new work but not the refactor

**Comment by marksvc** [comment]
@irahopkinson  This PR is back to being ready for review.

**Comment by marksvc** [comment]
Here is how I interpret what codecov is saying is missed or partial:
- The 2 partials and 2 misses regard situations that I am not aware of yet. If we see those warnings printed out we can pay attention to how we might have gotten into that situation and what we should be doing about it, if anything.

**Comment by marksvc** [comment]
@irahopkinson , Reviewable considers that the PR is still approved. That's not too unfortunate in this situation, but I think want a renewed approval after the recent rearrangements.

**Review by irahopkinson — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1179#-:-MpFSnF98Mnk_Igpl-3D:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @marksvc)

**Pull request merged**
